File: experiments/issue-282/parity/go/main.go
// Print what the Go implementation makes of the document.
package main

import (
	"fmt"
	"os"

	lino "github.com/link-foundation/links-notation/go"
)

func main() {
	document, err := os.ReadFile("../document.lino")
	if err != nil {
		panic(err)
	}
	links, err := lino.Parse(string(document))
	if err != nil {
		panic(err)
	}
	fmt.Println(lino.Format(links))
}

File: experiments/issue-282/parity/rust/src/main.rs
//! Print what the Rust implementation makes of the document.

fn main() {
    let document = include_str!("../../document.lino");
    let links = links_notation::parse_lino_to_links(document).expect("the document should parse");
    println!("{}", links_notation::format_links(&links));
}
